lib/rucio/daemons/tracer/kronos.py

```diff
@@ -341,87 +360,62 @@ def kronos_file(thread=0, dataset_queue=None, sleep_time=60):
     reconnect_attempts = config_get_int('tracer-kronos', 'reconnect_attempts')
     ssl_key_file = config_get('tracer-kronos', 'ssl_key_file', raise_exception=False)
     ssl_cert_file = config_get('tracer-kronos', 'ssl_cert_file', raise_exception=False)
-
-    sanity_check(executable=executable, hostname=hostname)
-    while not graceful_stop.is_set():
-        start_time = time()
-        heart_beat = live(executable, hostname, pid, hb_thread)
-        prepend_str = 'kronos-file[%i/%i] ' % (heart_beat['assign_thread'], heart_beat['nr_threads'])
-        logger = formatted_logger(logging.log, prepend_str + '%s')
-        conns = get_stomp_brokers(brokers=brokers_alias,
-                                  port=port,
-                                  use_ssl=use_ssl,
-                                  vhost=vhost,
-                                  reconnect_attempts=reconnect_attempts,
-                                  ssl_key_file=ssl_key_file,
-                                  ssl_cert_file=ssl_cert_file,
-                                  timeout=sleep_time,
-                                  logger=logger)
-        for conn in conns:
-            if not conn.is_connected():
-                logger(logging.INFO, 'connecting to %s' % str(conn.transport._Transport__host_and_ports[0]))
-                record_counter('daemons.tracer.kronos.reconnect.{host}', labels={'host': conn.transport._Transport__host_and_ports[0][0]})
-                conn.set_listener('rucio-tracer-kronos', AMQConsumer(broker=conn.transport._Transport__host_and_ports[0],
-                                                                     conn=conn,
-                                                                     queue=config_get('tracer-kronos', 'queue'),
-                                                                     chunksize=chunksize,
-                                                                     subscription_id=subscription_id,
-                                                                     excluded_usrdns=excluded_usrdns,
-                                                                     dataset_queue=dataset_queue,
-                                                                     bad_files_patterns=bad_files_patterns,
-                                                                     logger=logger))
-                if not use_ssl:
-                    conn.connect(username, password)
-                else:
-                    conn.connect()
-                conn.subscribe(destination=config_get('tracer-kronos', 'queue'), ack='client-individual', id=subscription_id, headers={'activemq.prefetchSize': prefetch_size})
-
-        tottime = time() - start_time
-        if tottime < sleep_time:
-            logger(logging.INFO, 'Will sleep for %s seconds' % (sleep_time - tottime))
-            sleep(sleep_time - tottime)
-
-    logger(logging.INFO, 'graceful stop requested')
+    conns = get_stomp_brokers(
```

Comment:
you cannot assign, this way it will not return the values to the calling context. Either update the array in place, or (probably better) use a dict `return_values` and assign the conns to it. Something like return_values['conns'] = get_stomp....